api/utils.go

```diff
@@ -0,0 +1,50 @@
+// @license
+// Copyright 2025 Dynatrace LLC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api/rest"
+)
+
+// ProcessResponse processes an HTTP response and applies a transformation function to the response body if provided.
+func ProcessResponse(httpResponse *http.Response, transformers ...func([]byte) ([]byte, error)) (Response, error) {
```

Comment:
Second, It's nowhere used with two transformers, so you move something in a function that is currently not needed twice.

I'd say, just do the error handling in this method (e.g. call `CreateResponse()`), and keep the transformation in the segments code. It's nowhere needed to introduce the transformation elsewhere. The segments code can transform the api.Response directly.

Reply:
rewritten